packages/core/src/engines/chat/DefaultContextGenerator.ts

```diff
@@ -68,15 +76,66 @@ export class DefaultContextGenerator
       message,
     );
 
-    // TODO: also use retrieved image nodes in context
+    const content: MessageContentDetail[] = [];
+    const nodeMap = splitNodesByType(nodes.map((r) => r.node));
+    for (const type in nodeMap) {
+      // for each retrieved modality type, create message content
+      const nodes = nodeMap[type as ModalityType];
+      if (nodes) {
+        content.push(
+          ...(await createMessageContent(
+            this.contextSystemPrompt,
+            type as ModalityType,
+            nodes,
+          )),
+        );
+      }
+    }
+
     return {
       message: {
-        content: this.contextSystemPrompt({
-          context: nodes.map((r) => (r.node as TextNode).text).join("\n\n"),
-        }),
-        role: "system",
+        content,
+        role: "user", // TODO: images in "system" messages don't work
```

Comment:
@logan-markewich @yisding system messages can't have images in OpenAI (see https://platform.openai.com/docs/api-reference/chat/create)
This means we can't add retrieved images to the context.
However the example (`examples/multimodal/context.ts`) works nicely with generating a user message as context.
Any ideas how to handle this?